### ISSUE TITLE
Show a user's cursor to collaborators when typing

### ISSUE THREAD

**Issue by Zsailer** [jupyterlab/jupyter-collaboration#261]
### Problem

In most collaborative editing applications, when a user is typing, their name appears over the cursor. This is a really helpful (and now expected UX) when collaborating in documents. This allows everyone to know who is making the current changes as they appear.

Jupyter collaboration doesn't do this. It only shows a name when you hover over the cursor. There are some improvements to that behavior we can make too, highlighted in #259.

### Proposed Solution

We should change Jupyter Collaboration to always show the user's name when they are typing.

### Additional context

See this video of Google docs as an example:

https://github.com/jupyterlab/jupyter-collaboration/assets/2791223/14127276-3b81-40e5-9407-77ddfa1585b8

**Comment by coolsyntax** [comment]
Hi, I would like to work on this issue. Please assign it to me.

**Comment by davidbrochart** [comment]
Thanks for offering your help @coolsyntax!

**Comment by coolsyntax** [comment]
Hi, I am working on this issue, but when I try to use the showTooltip function to create a state field and layer for the username tooltips, the usernames do not appear. I've tried creating a custom view plugin which updates the state on keyboard events, I've also tried creating a Widget decoration to display the username directly above the cursor, and looked at the documentation of CodeMirror, but failed to make any progress. Could you please help me solve this issue? I'd appreciate any guidance or feedback you can provide on the best way to display the user's name above their cursor while they are typing in the collaborative Jupyter environment.

**Comment by krassowski** [comment]
@coolsyntax can you open a draft PR with what you tried and think was the best attempt so far?

**Comment by arjxn-py** [comment]
Hi @davidbrochart @Zsailer @krassowski, I'd like to work on this issue. We've recently improved the UI of collaborator cursor on [JupyterGIS](https://github.com/geojupyter/jupytergis) in https://github.com/geojupyter/jupytergis/pull/1848 - we're planning to do something similar for notebooks too.

**Comment by davidbrochart** [comment]
Thanks a lot @arjxn-py, I've assigned you.
Correct me if I'm wrong, but I think JupyterGIS shows the *mouse* cursor, not the textual cursor as mentioned in this issue.

**Comment by arjxn-py** [comment]
Yes, JupyterGIS draws the mouse pointer on the map, not a text cursor. But I'll take the UI inspiration from there - not the mechanism.

**Comment by arjxn-py** [comment]
This can be closed now!
